### DISCUSSION TITLE
Why `ExecuteRemoteCommand` required `ShellOutputReceiver` to work well in specific jobs?

### DISCUSSION THREAD

**Discussion by syail** [yungd1plomat/AdvancedSharpAdbClient#116] [Q&A] [answered]
I tried to copy file local to local command and copy into my PC.

Example)
```cs
_client.ExecuteRemoteCommand($"su -c 'cp /data/local/tmp/test /sdcard/test'", device);

PullFile(device, $"/sdcard/test", "test");
```
In the above code, if there is no ```ShellOutputReceiver```, the copy does not seem to proceed, and I am curious as to why this is.

**Comment by syail** [comment]
I am guessing that the above problem is due to the code below.

https://github.com/SharpAdb/AdvancedSharpAdbClient/blob/77db1e933f98ac4562574d407eb5ee735f992211/AdvancedSharpAdbClient/AdbClient.cs#L389

If anyone knows the exact cause, please let me know

**Comment by wherewhere** [comment, marked as answer]
Maybe fixed this.
Read to end when ExecuteServerCommand without receiver https://github.com/SharpAdb/AdvancedSharpAdbClient/commit/b56b985583cee3cd4af7248c474e035e978df9ad
